driver/src/mongo_db.rs

```diff
@@ -84,36 +79,56 @@ impl DbInterface for MongoDB {
         }
         Ok(models::AccountState::from(docs.pop().unwrap()))
     }
+}
+
+impl DbInterface for MongoDB {
+    fn get_balances_for_state_root(
+        &self,
+        state_root: &H256,
+    ) -> Result<models::AccountState, DatabaseError> {
+        let query = doc!{ "stateHash" => format!("{:x}", state_root) };
+        info!("Querying stateHash: {}", query);
+        self.get_balances_for_query(query)
+    }
+
+    fn get_balances_for_state_index(
+        &self,
+        state_index: &U256,
+    ) -> Result<models::AccountState, DatabaseError> {
+        let query = doc!{ "stateIndex" => state_index.low_u64() };
```

Comment:
I'm curious, about this part.

I understand this would panic if the `state_index` don't fit in a u64. If this is the case, shouldn't it be u64 the argument type, so this restriction is part of the signature?

Reply:
The documentation just says

> Conversion to u64

I assume it takes to lowest 64 bits and truncates the rest (as in it doesn't panic if it overflows). The problem here is that in mongoDb we are storing state_index as a number and it can only represent up to 64 bits. For account balances we are using strings, which we should have used here too.
However, given that mongoDb will die at the end of this series of PRs it's probably not worth changing the schema.

Reply:
Yes, it's no worth changing it.

And I think you are also right about the panic, I didn't look it ok, I took a quick look and saw the panic from the next method in the docs:

![image](https://user-images.githubusercontent.com/2352112/63031001-99c8ae00-beb3-11e9-8269-eda14fdfa432.png)